unstructured_ingest/v2/processes/connectors/kafka.py

```diff
@@ -0,0 +1,197 @@
+
+from pydantic import Field, Secret, SecretStr
+
+from unstructured_ingest.error import (
+    SourceConnectionError,
+    SourceConnectionNetworkError,
+)
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    download_responses,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+if TYPE_CHECKING:
+    from confluent_kafka import Consumer
+
+CONNECTOR_TYPE = "kafka"
+
+
+class KafkaAccessConfig(AccessConfig):
+    api_key: Optional[SecretStr] = Field(
+        description="Kafka API key to connect at the server", alias="kafka_api_key", default=None
+    )
+    secret: Optional[SecretStr] = Field(description="", default=None)
+
+
+class KafkaConnectionConfig(ConnectionConfig):
+    access_config: Secret[KafkaAccessConfig] = Field(
+        default=KafkaAccessConfig(), validate_default=True
+    )
+    timeout: Optional[float] = 1.0
+    confluent: Optional[bool] = False
+    bootstrap_server: str
+    port: str
+    topic: str
+    num_messages_to_consume: Optional[int] = 1
+
+    @requires_dependencies(["confluent_kafka"], extras="kafka")
+    def get_client(self) -> "Consumer":
+        from confluent_kafka import Consumer
+
+        is_confluent = self.confluent
+        bootstrap = self.bootstrap_server
+        port = self.port
+
+        conf = {
+            "bootstrap.servers": f"{bootstrap}:{port}",
+            "client.id": socket.gethostname(),
+            "group.id": "your_group_id",
+            "enable.auto.commit": "false",
+            "auto.offset.reset": "earliest",
+            "message.max.bytes": 10485760,
+        }
+
+        if is_confluent:
+            api_key = self.access_config.kafka_api_key
+            secret = self.access_config.secret
+            conf["sasl.mechanism"] = "PLAIN"
+            conf["security.protocol"] = "SASL_SSL"
+            conf["sasl.username"] = api_key
+            conf["sasl.password"] = secret
+
+        consumer = Consumer(conf)
+        logger.debug(f"kafka consumer connected to bootstrap: {bootstrap}")
+        topic = self.topic
+        logger.info(f"subscribing to topic: {topic}")
+        consumer.subscribe([topic])
+        return consumer
+
+
+class KafkaIndexerConfig(IndexerConfig):
+    pass
+
+
+@dataclass
+class KafkaIndexer(Indexer):
+    connection_config: KafkaConnectionConfig
+    index_config: KafkaIndexerConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def _get_messages(self):
+        from confluent_kafka import KafkaError
+
+        consumer = self.connection_config.get_client()
+        running = True
+
+        collected = {}
+        num_messages_to_consume = self.connection_config.num_messages_to_consume
```

Comment:
I don't see this value being used anywhere

Reply:
Implemented a bit improved message read loop where I now properly use the value